java/src/main/java/com/google/appengine/tools/pipeline/impl/PipelineManager.java

```diff
@@ -159,6 +159,12 @@ public JobRecord registerNewJobRecord(UpdateSpec updateSpec, JobSetting[] settin
     // --> JobRecordFactory or something, that gets injected
     String projectId = backEnd.getOptions().as(AppEngineBackEnd.Options.class).getProjectId();
 
+    // if projectId is the local gae 'no_app_id', change it (new Datastore client blows up if attempt to create keys with it)
+    if (projectId.equals("no_app_id")) {
+      throw new IllegalStateException("projectId is 'no_app_id'; this isn't legal GCP project id, so changing to 'local-gae-project'");
+    }
+    projectId = "local-gae";
```

Comment:
```suggestion
    projectId = "local-gae-project";
```
for consistency

Reply:
And maybe better as a constant, so it can be reused in `RequestUtils` and here